.github/workflows/aws_deploy.yml

```diff
@@ -75,17 +82,30 @@ jobs:
 
       - name: Get timestamp
         id: timestamp
-        run: echo "timestamp=$(date +%Y%m%d%H%M%S)" >> $GITHUB_ENV
+        run: echo "timestamp=$(date +%Y%m%d%H%M%S)" >> $GITHUB_OUTPUT
+
+      - name: Checkout at release tag
+        uses: actions/checkout@v3
+        if: inputs.environment == 'production'
+        with:
+          ref: ${{ inputs.release_tag }}
+
+      - name: Get tags
+        id: tags
+        run: |
+          echo "commit-tag=$(if [[ ${{ inputs.environment }} == 'production' ]]; then echo $(git log -1 '--format=format:%H'); else echo ${{ github.sha }}; fi)" >> $GITHUB_OUTPUT
```

Comment:
`git log -1 '--format=format:%H'` is the best way I found for getting the commit hash for where HEAD is (i.e. at the release tag).

Reply:
It is, but `github.sha` refers to the commit that triggered the workflow, not where HEAD is right now. The full list of github contexts can be seen [here](https://docs.github.com/en/actions/learn-github-actions/contexts#github-context) - I don't think any of them are what we want, and the checkout action used further up the file doesn't output the commit sha (I saw a thread about this, in which people suggested what I did above as a way to fetch the current commit sha).

Reply:
Is the commit that triggered the workflow not the one we want? This is triggered on a release, but looking [here](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#release) it seems like `github.sha` should be the latest commit in the tagged release.

Reply:
Fair enough - I think because I wasn't testing this with a genuine release (e.g. I was forcing the 'prod' version of aws deploy workflow to run when pushing to a branch), the commit that triggered this workflow wasn't the commit of the release tag. But in a true deployment it would be, so `github.sha` would be appropriate in that case.